### PR TITLE
Execution module network support for AIX

### PR DESCRIPTION
**Pull request by dmurphy18**
### What does this PR do?
Provides full support on AIX for execution module network functionality

### What issues does this PR fix or reference?
https://github.com/saltstack/zh/issues/1192

### Previous Behavior
Not all of network functionality was functioning on AIX, as detailed in PR above

### New Behavior
Full functionality for network module on AIX, as detailed in PR above

### Tests written?
No - hand tested

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

File: salt/modules/network.py [hunk 1]
```diff
@@ -729,8 +843,15 @@ def arp():
             if comps[0] == 'Host' or comps[1] == '(incomplete)':
                 continue
             ret[comps[1]] = comps[0]
+        elif __grains__['kernel'] == 'AIX':
+            if comps[0] == 'bucket':
+                continue
+            if comps[0] == 'There':
```
Comment (cachedout) [inline salt/modules/network.py]: This could be collapsed into `if comps[0] in ('bucket', 'There'):`

File: salt/modules/network.py [hunk 2]
```diff
@@ -1350,7 +1479,7 @@ def default_route(family=None):
         default_route['inet'] = ['0.0.0.0', 'default']
         default_route['inet6'] = ['::/0', 'default']
     elif __grains__['os'] in ['FreeBSD', 'NetBSD', 'OpenBSD', 'MacOS', 'Darwin'] or \
-        __grains__['kernel'] == 'SunOS':
+        __grains__['kernel'] == 'SunOS' or __grains__['kernel'] == 'AIX':
```
Comment (cachedout) [inline salt/modules/network.py]: Same here. This could be written as `__grains__['kernel'] in ('SunOS', 'AIX'):`

File: salt/modules/network.py [hunk 3]
```diff
@@ -1466,6 +1598,39 @@ def get_route(ip):
...
+        ret = {
+            'destination': ip,
+            'gateway': None,
+            'interface': None,
+            'source': None
+        }
+
+        for line in out.splitlines():
+            line = line.split(':')
+            if 'route to' in line[0]:
```
Comment (cachedout) [inline salt/modules/network.py]: What happens here if no route is present?

**Review by cachedout — CHANGES_REQUESTED** [review]

File: salt/modules/network.py (hunk 1 above)
Comment (dmurphy18) [inline salt/modules/network.py]: will do

File: salt/modules/network.py (hunk 2 above)
Comment (dmurphy18) [inline salt/modules/network.py]: will do

File: salt/modules/network.py (hunk 3 above)
Comment (dmurphy18) [inline salt/modules/network.py]: On AIX if the route does not exist, it still outputs the commented out guide at line# 1602,

root@la68pp002_pub:/srv/pillar# route -n get 192.168.0.1   
   route to: 192.168.0.1
destination: default
       mask: default
    gateway: 172.29.128.13
  interface: en0
interf addr: 172.29.149.95
      flags: <UP,GATEWAY,DONE>
 recvpipe  sendpipe  ssthresh  rtt,msec    rttvar  hopcount      mtu     expire
       0         0         0         0         0         0         0   -315505 

If bad values are input then 
root@la68pp002_pub:/srv/pillar# route -n get 123.321.234.234
0821-287 123.321.234.234: bad value

And the pre-populated ret is returned, similar to the code for the other platforms, e.g.. OpenBSD, Solaris, etc.

**Review by cachedout — APPROVED** [review]

**Pull request merged**